components/dashboard/src/components/PendingChangesDropdown.tsx

```diff
@@ -4,12 +4,56 @@
  * See License.AGPL.txt in the project root for license information.
  */
 
-import { WorkspaceInstance } from "@gitpod/gitpod-protocol";
+import { WorkspaceInstance, WorkspaceInstanceRepoStatus } from "@gitpod/gitpod-protocol";
 import ContextMenu, { ContextMenuEntry } from "./ContextMenu";
 import CaretDown from "../icons/CaretDown.svg";
+import { useQuery } from "@tanstack/react-query";
+
+interface GitStatusResult {
+    status?: WorkspaceInstanceRepoStatus;
+}
+
+const useGitStatusQuery = ({ workspaceInstance }: { workspaceInstance?: WorkspaceInstance }) => {
+    return useQuery<GitStatusResult>({
+        queryKey: [
+            "git-status",
+            workspaceInstance?.id ?? "",
+            workspaceInstance?.status.phase ?? "",
+            workspaceInstance?.ideUrl ?? "",
+        ],
+        queryFn: async () => {
+            if (!workspaceInstance) {
+                return {};
+            }
+            if (workspaceInstance.status.phase !== "running") {
+                return {
+                    status: workspaceInstance.status.repo,
+                };
+            }
+            const workspaceUrl = new URL(workspaceInstance.ideUrl);
+            workspaceUrl.pathname = "/_supervisor/v1/status/git";
+            try {
+                const response = await fetch(workspaceUrl.toString(), { credentials: "include" });
+                if (response.status === 200) {
+                    const result = await response.json();
+                    return result;
+                }
+            } catch (e) {
+                console.debug("failed to fetch git status:", e);
+                // no-op, refetch each 5s
+            }
+            return {
+                // undefined, since we don't know state of running workspace
```

Comment:
As fallback, can we return `workspaceInstance.status.repo` so that at least something is shown?

Reply:
I don't think it is good idea for running workspace. On purpose removed since it will show some outdated thing.

Reply:
@gtsiolis wdyt? What should we show when workspace is running? I think showing old branch is going to be wrong for instance, but maybe we should show an error or something?

Reply:
Good point. Although I think it's not the right solution, I had an idea of showing a little warning icon with the outdated information and just letting the user know that it's not latest

Reply:
Yes, branch is part of git status, it should be up-to-date too. It is about there we should show info from last stop if we cannot resolve from running workspace, like right now, although it can be misleading.

Reply:
> ... if we cannot resolve from running workspace

Ah, I see. 💭 

- We could show something like _Couldn't fetch git status_ in place, if that's what happened.
- Alternatively, showing a loading state with a message like  _Fetching git status ..._ or a skeleton loader as we do with the workspace options (see [relevant discussion](https://gitpod.slack.com/archives/C042X6P86BZ/p1689157949753929) (internal))  could suffice, if we are actively trying to retrieve git status.